### PR TITLE
Add auth documentation

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

**Comment by EricWittmann** [comment]
Waiting on review by @smccarthy-ie before merging this.

File: docs/modules/ROOT/partials/getting-started/con-registry-overview.adoc [hunk 1]
```diff
@@ -45,3 +45,6 @@ endif::[]
 * Compatibility with existing Confluent schema registry client applications
 
 * Operator-based installation of {registry} on OpenShift
+
+* Support for oidc based authentication
+
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-overview.adoc:50]: Suggest spelling out the acronym:
"Support for OpenID Connect (OIDC) authentication"

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Metadata created by nebel
+
+[id="registry-security"]
+
+= Configuring {registry} with Authentication
+
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:6]: Suggest lowercase "authentication" for consistency with sentence style headings

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+This section explains how to configure authentication and authorization for {registry}. In order to configure the authentication the following environment variables should be used:
+
+.Environment variables for {registry} authentication configuration.
+//[%header,cols="5,5,2,5"]
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:10]: Suggest removing full stop from heading

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+|Name
+|Description
+|Type
+|Default
+|`AUTH_ENABLED`
+|If set to true, then the other environment variables are required.
+|String
+|`false`
+|`KEYCLOAK_URL`
+|The url of the auth server to be used. Must end with `/auth.
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:22]: Suggest: Must end with `/auth` (add closing  backtick)

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+|The client id to be used by the user interface
+|String
+|`apicurio-registry`
+|===
+
+By default, {registry} provides some default roles and configuration that are described below.
+
+
+.Environment variables for {registry} authentication configuration.
+//[%header,cols="5,5,2,5"]
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:43]: Suggest heading such as:
Default user roles for {registry} authentication

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+|`KEYCLOAK_URL`
+|The url of the auth server to be used. Must end with `/auth.
+|String
+|
+|`KEYCLOAK_REALM`
+|The realm to be used.
+|String
+|
+|`KEYCLOAK_API_CLIENT_ID`
+|The client id for the rest API.
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:30]: Suggest "The client ID for the REST API."

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 7]
```diff
@@ -0,0 +1,62 @@
+// Metadata created by nebel
+
+[id="registry-security"]
+
+= Configuring {registry} with Authentication
+
+This section explains how to configure authentication and authorization for {registry}. In order to configure the authentication the following environment variables should be used:
+
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:8]: Suggest "This section explains how to configure authentication and authorization for {registry} based on OpenID Connect (OIDC)."

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+|String
+|
+|`KEYCLOAK_API_CLIENT_ID`
+|The client id for the rest API.
+|String
+|`registry-api`
+|`KEYCLOAK_UI_CLIENT_ID`
+|The client id to be used by the user interface
+|String
+|`apicurio-registry`
```
Comment (smccarthy-ie) [inline docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:36]: Suggest "The client ID for the {registry} user interface"

File: docs/modules/ROOT/partials/getting-started/con-registry-security.adoc (hunk 5 above)
Comment (carlesarnal) [reply docs/modules/ROOT/partials/getting-started/con-registry-security.adoc:43]: Copy-paste mistake, sorry.

**Comment by EricWittmann** [comment]
@carlesarnal All set on this?

**Comment by carlesarnal** [comment]
> @carlesarnal All set on this?

Yes, @smccarthy-ie comments addressed.

**Pull request merged**